llvm/tools/sycl-post-link/CompileTimePropertiesPass.h

```diff
@@ -26,6 +28,19 @@ class CompileTimePropertiesPass
   // Enriches the module with metadata that describes the found variables for
   // the SPIRV-LLVM Translator.
   PreservedAnalyses run(Module &M, ModuleAnalysisManager &MAM);
+
+private:
+  // Transforms llvm.ptr.annotations with the "sycl-properties" annotation
+  // string into another llvm.ptr.annotations call in a format consumable by
+  // the SPIR-V translator.
+  bool transformSYCLPropertiesAnnotation(
+      Module &M, IntrinsicInst *IntrInst,
+      SmallVector<IntrinsicInst *, 4> &RemovableAnnotations);
+
+  // Map for keeping track of global variables generated for annotation strings.
+  // This allows reuse for annotations with the same generated annotation
+  // strings.
+  std::unordered_map<std::string, GlobalVariable *> NewAnnotationStrings;
```

Comment:
I cannot find a use of the `NewAnnotationStrings` member outside of the `transformSYCLPropertiesAnnotation` member function. Should `NewAnnotationString` be extracted as a class member or it can just be a local variable in the `transformSYCLPropertiesAnnotation` function? Do you have any intent to reuse the variable?

Reply:
The intention of having it as a class member is to share it between calls to `transformSYCLPropertiesAnnotation`, that is if two annotations need the same new annotation string, they can use the same global variable.

Reply:
@steffenlarsen Thank you for the explaining. I get, I missed the fact that the variable is not cleared before exit from the function, so this variable is shared between calls.